smolyay/basis.py

```diff
@@ -26,6 +26,12 @@ def domain(self):
         """numpy.ndarray: Domain the sample points come from."""
         pass
 
+    @property
+    @abc.abstractmethod
+    def integral_over_domain(self):
+        """float: basis function integrated over the domain."""
+        pass
```

Comment:
I will put a suggestion below to call the method `integrate()` in the surrogates, and so I think the basis functions should also use that.
```suggestion
    @abc.abstractmethod
    def integrate(self):
        """Integrate basis function over its domain."""
        pass
```
It would make sense to move this method after the `derivative` method as well.

Revise as needed elsewhere based on this change.